packages/agreement-process/src/model/domain/errors.ts

```diff
@@ -0,0 +1,97 @@
+import { ApiError, DescriptorState } from "pagopa-interop-models";
+
+export function eServiceNotFound(eServiceId: string): ApiError {
+  return new ApiError({
+    detail: `EService ${eServiceId} not found`,
+    code: "0007",
+    httpStatus: 400,
```

Comment:
the exception should not be bound to the status code.
For example, both `createAgreement` and `submitAgreement` raise an `eServiceNotFound` error, but the first returns `400` while the second returns `500`